frontend/src/components/modals/settings/AutocompleteCombobox.tsx

```diff
@@ -43,6 +44,14 @@ export function AutocompleteCombobox({
 }: AutocompleteComboboxProps) {
   const { t } = useTranslation();
 
+  const [selectedKey, setSelectedKey] = React.useState<Key>(defaultKey);
+  React.useEffect(() => {
+    if (!allowCustomValue) setSelectedKey(defaultKey);
+    else {
+      setSelectedKey("");
+    }
+  }, [defaultKey]);
+
```

Comment:
If `allowCustomValue` is false, set the `selectedKey` to the `defaultKey`. If `allowCustomValue` is true and the `defaultKey` is not in the dropdown list (i.e., a custom value), set the `selectedKey` to an empty string. This ensures that when the user types a custom value, the input field is not pre-populated with the `defaultKey` if it's not in the dropdown list. It also removes the checkmark on the `selectedKey` if the user starts typing in a field where a custom value is allowed.